libraries/ts-command-line/src/providers/CommandLineParameterProvider.ts

```diff
@@ -832,16 +866,58 @@ export abstract class CommandLineParameterProvider {
           throw new Error('Unexpected parameter group: ' + parameterGroup);
         }
 
-        argumentGroup = this._getArgumentParser().addArgumentGroup({
+        argumentGroup = argumentParser.addArgumentGroup({
           title: `Optional ${parameterGroupName} arguments`
         });
         this._parameterGroupsByName.set(parameterGroup, argumentGroup);
       }
     } else {
-      argumentGroup = this._getArgumentParser();
+      argumentGroup = argumentParser;
     }
 
-    argumentGroup.addArgument(names, { ...argparseOptions });
+    const argparseArgument: argparse.ArgumentOptions = (argumentGroup as IExtendedArgumentGroup).addArgument(
+      names,
+      argparseOptions
+    );
+    if (required && environmentVariable) {
+      // Add some special-cased logic to handle required parameters with environment variables
+      parameter._preParse = () => {
+        // Set the value as non-required before parsing. We'll validate it explicitly
+        argparseArgument.required = false;
+      };
+      parameter._postParse = () => {
+        // Reset the required value to make the usage text correct
+        argparseArgument.required = true;
+      };
```

Comment:
These should probably call the original methods before performing the additional work